# Introduction to Project

## Project Description

For this project you must create a data set by simulating a real-world phenomenon of
your choosing. You may pick any phenomenon you wish – you might pick one that is
of interest to you in your personal or professional life. Then, rather than collect data
related to the phenomenon, you should model and synthesise such data using Python.
We suggest you use the numpy.random package for this purpose.
Specifically, in this project you should:

- Choose a real-world phenomenon that can be measured and for which you could collect at least one-hundred data points across at least four different variables.
- Investigate the types of variables involved, their likely distributions, and their relationships with each other.
- Synthesise/simulate a data set as closely matching their properties as possible.
- Detail your research and implement the simulation in a Jupyter notebook the data set itself can simply be displayed in an output cell within the notebook.

Note that this project is about simulation – you must synthesise a data set. Some students may already have some real-world data sets in their own files. It is okay to base your synthesised data set on these should you wish (please reference it if you do),
but the main task in this project is to create a synthesised data set.

## Main Sources Used For Project

- Python[1] is the main code used for project 

- Seaborn[2] is a library used for making attractive and informative statistical graphics in Python.

- Pandas[3] are used for data manipulation and analysis.

- Numpy.Random [4] is a subpackage of the NumPy package for working with random numbers.

## Aim for project

The aim for my project is to look at military spending by countries for 2021. I firstly looked at World Population Review[5] I wanted to see real life data and what research had been done in this area already. A lot of the information, has been based on research by Stockholm International Peace Research Institute(SIPRI Military Expenditure Database)[6] as a basis for a lot of my research. 

The main variables which I will research are the population of the country, how much they are spending on their military, country's GDP and finally the percentage of GDP that makes up the countries spending per year. 

As of 2021, world military expenditure has passed $2 trillion for first time. Total global military expenditure increased by 0.7 per cent in real terms in 2021, to reach $2113 billion with the 5 biggest countries being the US, China, India, the UK and Russia.[3]

In [ ]:
# import python libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

I firstly decided to look at real life data. I had to very slightly adapt the database which I got from SIPRI. I removed the first approximately 10 lines on each excel sheet as I was finding it difficult read in the data. The main pieces that I removed provided just general info on the database.

In [3]:
import pandas as pd

df = pd.read_excel(r'C:\Users\Kenne\OneDrive\Desktop\Programming\PfDA_Project\data\SIPRI DATA.xlsx', sheet_name= 'Current US')

GDP = pd.read_excel(r'C:\Users\Kenne\OneDrive\Desktop\Programming\PfDA_Project\data\SIPRI DATA.xlsx', sheet_name= 'Share of GDP')

percentage = pd.read_excel(r'C:\Users\Kenne\OneDrive\Desktop\Programming\PfDA_Project\data\SIPRI DATA.xlsx', sheet_name= 'Share of Govt. spending')

print("The shape of the data from Current US$ is: \n",df.shape)

print("The shape of the data from GDP is: \n",GDP.shape)

print("The shape of the data from percentage of spending is: \n",percentage.shape)

The shape of the data from Current US$ is: 
 (191, 75)
The shape of the data from GDP is: 
 (191, 75)
The shape of the data from percentage of spending is: 
 (191, 37)


I decided to look at the shape of the data firstly and it shows that there 191 columns in the database, mostly displaying countries and continents in which they are in. There are also 75 rows of data which displays the amount spent by a country, in dollars. The GDP data on the dataset is the exact same in terms of columns and rows.

I also wanted to dipslay that my idea for the 

In [14]:
df = pd.read_excel(r'C:\Users\Kenne\OneDrive\Desktop\Programming\PfDA_Project\data\SIPRI DATA.xlsx', sheet_name= 'Current US')

df.isnull().any(axis = 1).sum()

56

I saw a bit of an issue with dataset as there was a lot of missing data. 

If I was to ever do further analysis on this dataset, I think I would take specific years as opposed to all the data.

## 2.Investigate the types of variables involved, their likely distributions, and their relationships with each other

In this part of the assignment, I am going to create the dataset based on numpy random and begin to investigate distributions and relationships with one another. 

Firstly, I will investigate the variables in the datasets used by SIPRI. I will use histograms and boxplots to show some of the variables in this dataset. I will then explore relationships that may exist between the variables using visualisations such as scatterplot, pairplots etc and statistics such as correlation and covariance statistics.

I will use this to create a synthesised data set.

I have decided to use the actual country names, just purely as I find it easier to analyse with a real name as opposed to a pseudo name. 

I will create the country's GDP, total spending and percentages of spending using random numpy.

I will then use this data to visualise relationships which we may find.